### ISSUE TITLE
Attaching other label/text to a connection

### ISSUE THREAD

**Issue by posenato** [bpmn-io/bpmn-js#1790]
__Describe the problem__

In some contexts, it is important to have the possibility to attach a small text to a connection.

For example, a label like '[1,10]' for representing the allowed time cost range of the connection.

Currently (bpmn-js v11.0.5), it seems only possible to attach a label connected to the name of the connection.

Even customizing `CustomRenderer.prototype.drawConnection()`, it is not possible to have a simple SVG text attached to the connection.

I discovered the reason, and I published a post where I described it. Briefly, after the preview of a connection, only polyline elements are rendered in the browser.

For shape, the problem is not present. See the below figure.

The post is in the bpmn.io forum: [[How to add a second label to a connection](https://forum.bpmn.io/t/how-to-add-a-second-label-to-a-connection/8414)](https://forum.bpmn.io/t/how-to-add-a-second-label-to-a-connection/8414)

__Describe the solution you'd like__

A simple solution would be to allow all SVG elements created in `CustomRenderer.prototype.drawConnection()` to be rendered, not only SVG polyline elements.

A more high-level solution would be to offer a method for creating labels (each associated with a BusinessObject property) asking only for the name of the property and the position (one of the predefined set of positions) where to put the text.

__Describe alternatives you've considered__

For now, I solved the problem using the overlays. See the below figure.

__Figure__

<img width="343" alt="ScreenShot 2022-12-09 at 11 45 50" src="https://user-images.githubusercontent.com/7280315/206695748-8741102a-29b6-4e1d-acde-9c885f59a9a6.png">

**Comment by nikku** [comment]
@posenato Thanks for opening this issue.

Labels on tasks are _internal_, labels on connectors (and events) are _external_.

You'd want to 

* explicitly [attach it](https://github.com/bpmn-io/bpmn-js/blob/develop/lib/import/BpmnImporter.js#L248)
* implement editing for it
* move the label with the connection (https://github.com/bpmn-io/bpmn-js/blob/develop/lib/features/modeling/behavior/LabelBehavior.js#L209)

**Comment by posenato** [comment]
@nikku  Thanks for your reply, although I don't catch the part

> You'd want to
> - explicitly [create an external connection label](https://github.com/bpmn-io/bpmn-js/blob/develop/lib/import/BpmnImporter.js#L248)
> - implement editing for it
> - [move the label with the connection](https://github.com/bpmn-io/bpmn-js/blob/develop/lib/features/modeling/behavior/LabelBehavior.js#L209)

I tried to read and use BpmnImporter.js, but it is not clear how to use it, and I couldn't create an external label (that should be a fixed text label and not be editable by users). 

Is there any documentation on how to set/options for external labels?

Thanks.

**Comment by nikku** [comment]
@posenato You say it is fixed, but how shall it behave if the connection re-layouts.

What I did in https://github.com/bpmn-io/bpmn-js/issues/1790#issuecomment-1346270764 is suggest how to add another connection label / text, in the proper manner.

If that is not what you need, could you elaborate what you need? How shall the label behave:

* During connection creation
* Connection change (re-connect), ...

If you need a solution for "view only" that may be a different one.

**Comment by nikku** [comment]
@posenato To manage expectations, [my comment](https://github.com/bpmn-io/bpmn-js/issues/1790#issuecomment-1346270764) did not imply it was "easy" to extend it, rather I outlined the high level steps. 

The BPMN editor is a complex piece of software and especially new use cases (multiple labels) may require additional work. You'd need to learn a bunch about how bpmn-js works (at the core) in order to understand what is going on. Did you checkout the [walkthrough](https://bpmn.io/toolkit/bpmn-js/walkthrough/#diagram-js) already for a high level overview? Unfortunately that one does not cover the import process in depth yet (https://github.com/bpmn-io/bpmn.io/issues/154).

I've sketched a TLDR of the high level import process (https://github.com/bpmn-io/bpmn.io/issues/154#issuecomment-1346852110). Unfortunately you'd need to resort to your own discovery tools (debugging and friends) to dive into the details.

From your requirements I'd argue you'd need to go the external label route. Alternatively, to hack things you can replace the BPMN renderer entirely, to suite your needs.

**Comment by nikku** [comment]
If you feel like rendering only would solve your issue, please [share a code sandbox](https://codesandbox.io/s/bpmn-modeler-6wnf7) that reproduces the issue. It may be a small fix on our end.

**Comment by posenato** [comment]
As I wrote in the first post, I implemented a pseudo-read-only label system for connection using an SVG:text in overlay.

I was only wondering if there is a more high-level way to realize such labels, but it seems that there isn't.

Just a final comment. 
I don't catch why for shapes, the system copies (from the preview) all the SVG elements present in the SVG parent of the element, while for connections, it copies only the SVG polylines.

(I verified: if during a connection rendering, I add different SVG:polylines to the SVG parent, all such polylines are rendered, while if I add an SVG:text, such text is not rendered.)
